src/nativestore/CentralNodeBlock.cpp

```diff
@@ -0,0 +1,290 @@
+                  std::to_string(currentRelation->addr);
+        }
+        if (!node) {
+            throw "Error creating node in the relation";
+        }
+        edges.push_back(*node);
+    }
+    return edges;
+}
+
+std::map<std::string, char*> CentralNodeBlock::getAllProperties() {
+    std::map<std::string, char*> allProperties;
+    CentralPropertyLink* current = this->getPropertyHead();
+    while (current) {
+        allProperties.insert({current->name, current->value});
+        CentralPropertyLink* temp = current->next();
+        delete current;  // To prevent memory leaks
+        current = temp;
+    }
+    delete current;
+    return allProperties;
+}
+
+CentralNodeBlock* CentralNodeBlock::get(unsigned int blockAddress) {
+    CentralNodeBlock* CentralNodeBlockPointer = NULL;
+    CentralNodeBlock::centralNodesDB->seekg(blockAddress);
+    unsigned int nodeId;
+    unsigned int edgeRef;
+    unsigned int centralEdgeRef;
+    unsigned char edgeRefPID;
+    unsigned int propRef;
+    char usageBlock;
+    char label[CentralNodeBlock::LABEL_SIZE];
+    std::string id;
+
+    if (!CentralNodeBlock::centralNodesDB->get(usageBlock)) {
+        central_node_block_logger.error("Error while reading usage data from block " + std::to_string(blockAddress));
+    }
+    if (!CentralNodeBlock::centralNodesDB->read(reinterpret_cast<char*>(&nodeId), sizeof(unsigned int))) {
+        central_node_block_logger.error("Error while reading nodeId  data from block " + std::to_string(blockAddress));
+    }
+    if (!CentralNodeBlock::centralNodesDB->read(reinterpret_cast<char*>(&edgeRef), sizeof(unsigned int))) {
+        central_node_block_logger.error("Error while reading edge reference data from block " + std::to_string(blockAddress));
+    }
+    if (!CentralNodeBlock::centralNodesDB->read(reinterpret_cast<char*>(&centralEdgeRef), sizeof(unsigned int))) {
+        central_node_block_logger.error("Error while reading central edge reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!CentralNodeBlock::centralNodesDB->read(reinterpret_cast<char*>(&edgeRefPID), sizeof(unsigned char))) {
+        central_node_block_logger.error("Error while reading edge reference partition ID data from block " +
+                                std::to_string(blockAddress));
+    }
+
+    if (!CentralNodeBlock::centralNodesDB->read(reinterpret_cast<char*>(&propRef), sizeof(unsigned int))) {
+        central_node_block_logger.error("Error while reading prop reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!CentralNodeBlock::centralNodesDB->read(&label[0], CentralNodeBlock::LABEL_SIZE)) {
+        central_node_block_logger.error("Error while reading label data from block " + std::to_string(blockAddress));
+    }
+    bool usage = usageBlock == '\1';
+    central_node_block_logger.debug("Label = " + std::string(label));
```

Comment:
Why is this line repeated twice?

`central_node_block_logger.debug("Label = " + std::string(label));`